core/shared/src/main/scala/fs2/Chunk.scala

```diff
@@ -20,7 +30,7 @@ import cats.implicits._
   * intermediate chunks being created (1 per call to `map`). In contrast, a chunk can be lifted to a segment
   * (via `toSegment`) to get arbitrary operator fusion.
   */
-abstract class Chunk[+O] extends Serializable {
+trait Chunk[@specialized(Byte, Short, Long, Char, Int, Float, Double) +O] extends Serializable {
```

Comment:
Unfortunately, scala didn't allow me to keep this as an abstract class.

Reply:
Changing to a trait means it will bbe harder to preserve backwards compatibility on 2.11. Not a deal breaker if the specialization pays off though.

Reply:
Hmm.

It's hard to say whether there's much of a benefit. Methods such as `apply` work better, but like `Stream`, `map` and other generic methods don't see any other benefit.

I'm leaning more of removing the specialization since `Stream` doesn't have it and I can't imagine most users are using `apply` very much and in the cases that matter using `toArray` or adding helper methods to get the underlying `Buffer` would serve the user better.

wdyt?

Reply:
My two-cents: `Chunk#apply` probably shouldn't exist at all if we can avoid it, and that's really the only place where the specialization would matter. In all other cases, users are forced to respecialize *anyway* in order to avoid boxing, so it shouldn't matter.